api/admin/controller.py

```diff
@@ -1440,48 +1440,32 @@ def _create_or_update_list(self, library, name, entries, collections, id=None):
         old_entries = [x for x in list.entries if x.edition]
         membership_change = False
         for entry in entries:
-            pwid = entry.get("pwid")
-            medium = entry.get("medium")
-            language = entry.get("language")
-            data_source = entry.get("data_source")
-
-            data_source = DataSource.lookup(self._db, data_source)
-            if data_source:
-                data_source_id = data_source.id
-            else:
-                data_source_id = None
+            urn = entry.get("identifier_urn")
 
+            identifier, ignore = Identifier.parse_urn(self._db, urn)
             query = self._db.query(
                 Work
             ).join(
                 Edition, Edition.id==Work.presentation_edition_id
+            ).join(
+                LicensePool, LicensePool.work_id==Work.id
+            ).join(
+                Collection, LicensePool.collection_id==Collection.id
             ).filter(
-                Edition.permanent_work_id==pwid
+                Edition.primary_identifier_id==identifier.id
+            ).filter(
```

Comment:
I'm not convinced it's necessary to join against Edition at all -- can you filter against LicensePool.identifier_id instead? That will be simpler and I think will be more resistant to edge cases (since a Work might consolidate LicensePools with a number of different identifiers).